r3bbase/R3BFileSource.cxx

```diff
@@ -400,10 +422,19 @@ Bool_t R3BFileSource::Init()
             std::string buf1 = buffer.c_str();
             buf1.erase(buf1.find(' '), buf1.length());
             buffer.erase(0, buffer.find(' ') + 1);
-            fexpid = strtol(buf1.c_str(), &p, 16);
-            ftimestamp[array_size] = strtol(buffer.c_str(), &p, 16);
-            LOG(DEBUG) << fexpid << " / " << ftimestamp[array_size];
-            array_size++;
+            fExpid = strtol(buf1.c_str(), &p2, 16);
+            uint64_t ts = strtol(buffer.c_str(), &p3, 16);
+
+            if (buf1.c_str() == p2 || buffer.c_str() == p3)
+            {
+                LOG(ERROR) << "R3BFileSource::Init() Bad structure for RunId file, \n"
+                           << "it must be (hex format): \n"
+                           << "fRunID = a \n"
+                           << "expid timestamp \n";
+                return kFALSE;
+            }
+            fTimestamp.push_back(ts);
```

Comment:
I think it makes sense here, since you are using two vectors now, that you check after pushing the timestamp, that the sizes of fTimestamp and fRunid are equal. Otherwise the two vectors are running out of sync (perhaps due to some bug introduced in the future). Just a sanity check.

I've seen the sync check further down, but it is better to (also) check here in every iteration of the loop, because two errors (one in timestamp and one in runid) may cancel each other out.